features/nanostack/mbed-mesh-api/mbed-mesh-api/mesh_interface_types.h

```diff
@@ -64,6 +64,93 @@ typedef enum {
     MESH_DEVICE_TYPE_WISUN_BORDER_ROUTER        /*<! Wi-SUN border router */
 } mesh_device_type_t;
 
+/**
+ * Mesh network statistics
+ */
+typedef struct {
+    /* 6Lowpan */
+    uint32_t ip_rx_count;           /*<! IP RX packet count. */
+    uint32_t ip_tx_count;           /*<! IP TX packet count. */
+    uint32_t ip_rx_drop;            /*<! IP RX packet drops count. */
+    uint32_t ip_cksum_error;        /*<! IP checksum error count. */
+    /* IP Payload Flow */
+    uint32_t ip_tx_bytes;           /*<! IP TX bytes count. */
+    uint32_t ip_rx_bytes;           /*<! IP RX bytes count. */
+    uint32_t ip_routed_up;          /*<! IP routed UP bytes count. */
+    uint32_t ip_no_route;           /*<! IP no route count. */
+    /* Fragments */
+    uint32_t frag_rx_errors;        /*<! Fragmentation RX error count. */
+    uint32_t frag_tx_errors;        /*<! Fragmentation TX error count. */
+    /*RPL stats*/
+    uint32_t rpl_route_routecost_better_change; /*<! RPL parent change count. */
+    uint32_t ip_routeloop_detect;               /*<! RPL route loop detection count. */
+    uint32_t rpl_memory_overflow;   /*<! RPL memory overflow count. */
+    uint32_t rpl_parent_tx_fail;    /*<! RPL transmit errors to DODAG parents. */
+    uint32_t rpl_unknown_instance;  /*<! RPL unknown instance ID count. */
+    uint32_t rpl_local_repair;      /*<! RPL local repair count. */
+    uint32_t rpl_global_repair;     /*<! RPL global repair count. */
+    uint32_t rpl_malformed_message; /*<! RPL malformed message count. */
+    uint32_t rpl_time_no_next_hop;  /*<! RPL seconds without a next hop. */
+    uint32_t rpl_total_memory;      /*<! RPL current memory usage total. */
+    /* Buffers */
+    uint32_t buf_alloc;             /*<! Buffer allocation count. */
+    uint32_t buf_headroom_realloc;  /*<! Buffer headroom realloc count. */
+    uint32_t buf_headroom_shuffle;  /*<! Buffer headroom shuffle count. */
+    uint32_t buf_headroom_fail;     /*<! Buffer headroom failure count. */
```

Comment:
These are useless stats and We are missing more important stats that are available memory usage/max alloc fails

Reply:
removed